platform/engineering/request-for-comment/2022-05-05-mobile-route-naming-conventions.md

```diff
@@ -0,0 +1,201 @@
+# RFC: Mobile Routes naming conventions
+
+- Date: 2022-05-05
+- Related Issue: https://app.zenhub.com/workspaces/va-mobile---backend-61d75d03a7505500147ec7f6/issues/department-of-veterans-affairs/va-mobile-app/2937
+
+## Background
+Mobile routes currently have some inconsistencies in their naming conventions, such as:
+- Sometimes using snake_case over kebab-case (maintenance_windows, message_drafts)
+- :ids are after the resource action rather than before (canceling appointments)
+- Not nesting under parent products (the 'health' and 'benefits' app)
+
+## Motivation
+Keeping naming consistent will make using the Mobile API much easier to use and is generally a good practice for maintainability. 
+
+## Design
+Version all endpoint that do not meet the naming convention as follows:
+```
+namespace :health do
+  get '/appointment-requests/:appointment_request_id/messages', to: 'appointment_request_messages#index'
+  get '/appointments', to: 'appointments#index'
+  put '/appointments/:id/cancel', to: 'appointments#cancel'
+  get '/appointments/community-care/:service_type/eligibility', to: 'community_care_eligibility#show'
+  get '/appointments/va/eligibility', to: 'veterans_affairs_eligibility#show'
+  get '/appointments/facility/eligibility', to: 'facility_eligibility#index'
+  post '/appointment', to: 'appointments#create'
+  get '/community-care-providers', to: 'community_care_providers#index'
+  end
+
+namespace :benefits do
+  get '/appeal/:id', to: 'claims_and_appeals#get_appeal'
+  get '/claims-and-appeals-overview', to: 'claims_and_appeals#index'
+  get '/claim/:id', to: 'claims_and_appeals#get_claim'
+  post '/claim/:id/documents', to: 'claims_and_appeals#upload_document'
+  post '/claim/:id/documents/multi-image', to: 'claims_and_appeals#upload_multi_image_document'
+  post '/claim/:id/request-decision', to: 'claims_and_appeals#request_decision'
+  end
+
+  get '/maintenance-windows', to: 'maintenance_windows#index'
+  get '/push/:endpoint_sid/prefs', to: 'push_notifications#get_prefs'
+  put '/push/:endpoint_sid/prefs', to: 'push_notifications#set_pref'
+
+scope :messaging do
```

Comment:
Messaging is health; a user has to be registered as a patient at a VA health facility to use it. So I think we can flip the scopes so the path ends up being `health/messaging`.